jgrapht-core/src/main/java/org/jgrapht/alg/drawing/model/Rectangles.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * (C) Copyright 2018-2019, by Dimitrios Michail and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * See the CONTRIBUTORS.md file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the
+ * GNU Lesser General Public License v2.1 or later
+ * which is available at
+ * http://www.gnu.org/licenses/old-licenses/lgpl-2.1-standalone.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR LGPL-2.1-or-later
+ */
+package org.jgrapht.alg.drawing.model;
+
+import org.jgrapht.alg.util.Pair;
+
+/**
+ * A collection of utilities to assist with rectangles manipulation.
```

Comment:
Similar comment as with class Points regarding genericization.

Reply:
I made an attempt, but you need to starting defining operations like monoids and stuff which I would rather avoid. 

I initially added the <N> number type as a parameter so that we can implement algorithms using exact arithmetic.